attributegeneric.go

```diff
@@ -0,0 +1,41 @@
+package yoti
+
+//AttributeGeneric is a Yoti attribute which returns a generic value, which must be explicitly cast
+type AttributeGeneric struct {
```

Comment:
I do agree that GenericAttribute is better, but since all the files are under the same directory, I thought it would be useful if they were grouped together alphabetically, I think it makes it a little more organised. If we think that's not important at all then I can change them

Reply:
You can do that organization with a `attribute` directory, and placing all different attribute types in there.

Reply:
I can do that in every other language I know of, but as I understand it it's not possible to have sub-directories which are in the same package in Go, as Go maps package paths onto directories (non-recursively). It's not entirely clear from the [official Go guidelines](https://blog.golang.org/package-names) ("Package Paths" section), but it's mentioned on the [Package Clause](https://golang.org/ref/spec#Package_clause) of the Go spec. 

So I don't think putting them in a separate directory is an option

Reply:
But why does stuff need to be in the same package? if organizationally it feels like they should all be "related" and in the same dir, it probably also means from an API pov that they should probably be in their own package and export whatever methods they need to export.